src/VisualStudio/Core/Def/Implementation/Workspace/VisualStudioSymbolNavigationService.cs

```diff
@@ -182,6 +183,17 @@ public bool TryNavigateToSymbol(ISymbol symbol, Project project, OptionSet optio
             return true;
         }
 
+        public async Task<MetadataAsSourceFile?> GetAndOpenGeneratedFileAsync(ISymbol symbol, Project project, CancellationToken cancellationToken)
```

Comment:
Previously, invoking go-to-def on a metadata file on the client did not open the metadata file on the host or add it to the metadata workspace, both of which are required for the call [here](https://github.com/dotnet/roslyn/pull/48663/files#diff-c08b7a14d241149e7cb52de82e1c7ca615b77c44cc9f995f1f2faf193bc159ccR60) to successfully return documents.

Note that this change silently opens the file on the host, i.e. in the Liveshare scenario, the host may not even realize the metadata file has been opened as we do not visibly open the file (intentionally, via the lack of a call to  [windowFrame.Show()](https://github.com/dotnet/roslyn/pull/48663/files#diff-652587c024375552b1efd57a5f38dc88454a9c2c7895c799d1f388c785bf9b94R168)). This matches the current behavior when opening files on the client.

Reply:
Can we move a comment here into the code?